akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventSourcedBehaviorImpl.scala

```diff
@@ -193,6 +185,9 @@ private[akka] final case class EventSourcedBehaviorImpl[Command, Event, State](
     copy(recovery = Recovery(selection.toUntyped))
   }
 
+  override def snapshotWhen(predicate: (State, Event, Long) => Boolean): EventSourcedBehavior[Command, Event, State] =
+    copy(snapshotWhen = predicate)
+
```

Comment:
I still think this rather belongs in the retention criteria somehow, given that is where the snapshot-on-seq-nr lives.

Reply:
It's not possible to have it there because of the State and Event type parameters.

I think we can think of retention as the automatic snapshotting and deletions, while this is very domain specific snapshotting.

Reply:
Hmmm, ok.

Reply:
That's how I see it as well, configuration of automatic by criteria vs snapshotWhen = specific